src/include/distributed/shard_rebalancer.h

```diff
@@ -105,7 +105,7 @@ typedef struct PlacementUpdateEventProgress
 	int sourcePort;
 	char targetName[255];
 	int targetPort;
-	uint64 progress;
+	pg_atomic_uint64 progress;
```

Comment:
Just a note: there's also pg_atomic_uint32

Reply:
64 bits indeed seems a bit excessive for values 0-4, but I'll leave it as is since memory usage here is not something we need to worry about.